src/__swaps__/screens/Swap/components/GestureHandlerButton.tsx

```diff
@@ -62,40 +66,59 @@ export function GestureHandlerButton({
   pointerEvents = 'box-only',
   scaleTo = 0.86,
   style,
+  tapRef,
 }: GestureHandlerButtonProps) {
-  const gesture = useMemo(
-    () =>
-      Gesture.Race(
-        Gesture.Tap()
-          .enabled(!disabled)
-          .onBegin(e => {
-            if (onPressStartWorklet) onPressStartWorklet(e);
-          })
-          .onEnd(e => {
-            if (onPressWorklet) onPressWorklet(e);
-            if (onPressJS) runOnJS(onPressJS)(e);
-          }),
-        Gesture.LongPress()
-          .enabled(!disabled && !!(onLongPressEndWorklet || onLongPressJS || onLongPressWorklet))
-          .minDuration(longPressDuration)
-          .onStart(e => {
-            if (onLongPressWorklet) onLongPressWorklet(e);
-            if (onLongPressJS) runOnJS(onLongPressJS)(e);
-          })
-          .onFinalize((_, success) => {
-            if (onLongPressEndWorklet) onLongPressEndWorklet(success);
-          })
-      ),
-    [disabled, longPressDuration, onLongPressEndWorklet, onLongPressJS, onLongPressWorklet, onPressJS, onPressStartWorklet, onPressWorklet]
-  );
+  const gesture = useMemo(() => {
+    const tap = Gesture.Tap()
+      .enabled(!disabled)
+      .onBegin(e => {
+        if (onPressStartWorklet) onPressStartWorklet(e);
+      })
+      .onEnd(e => {
+        if (onPressWorklet) onPressWorklet(e);
+        if (onPressJS) runOnJS(onPressJS)(e);
+      });
+
+    if (tapRef) tap.withRef(tapRef);
+
+    const longPressEnabled = !!(onLongPressEndWorklet || onLongPressJS || onLongPressWorklet);
+
+    if (!longPressEnabled) return tap;
+
+    const longPress = Gesture.LongPress()
+      .enabled(!disabled && !!(onLongPressEndWorklet || onLongPressJS || onLongPressWorklet))
```

Comment:
```suggestion
      .enabled(!disabled && longPressEnabled)
```

Reply:
Also at this point we probably don't need to actually check for `longPressEnabled` since it can't make it to this block if it's not true.

Reply:
Good call